pyproject.toml

```diff
@@ -1,15 +1,15 @@
 [build-system]
 requires = [
-    "setuptools>=64",
-    "wheel",
+    "scikit-build-core>=0.10,<=0.10.3",
```

Comment:
For the same reason, scikit-build-core only provides experimental support for editable-rebuild installs.
1. To avoid unwanted breaking changes
2. It can help reproducibility if someone builds PyBaMM's older releases from the source sometime soon.

Reply:
Does it work without editable rebuilds? I am bringing up the version pinning because it needs to be changed in multiple places whenever we update that pin. Ideally you should control this stuff in a single location or have a flexible range (e.g. `>=0.10.0,<0.11.0`).

At the moment the version needs to be manually updated in the pyproject file, the docker file, the documentation, and the CI.

Reply:
Yes, you are right. It would be hard to keep track of the versions and update them. The normal editable installs (without rebuilds) work without any problem and don't depend on the `--no-build-isolation` flag too. 

We can remove the editable-rebuilds and keep only editable installation to remove the version pinning.